design/dedicated-osd-pod.md

```diff
@@ -23,22 +23,34 @@ A more comprehensive discussions can be found at [this issue](https://github.com
+
+- OSD Provisioner. A process that is given device or directory paths upon start and make backend specific storage types. For now, the process prepare Ceph OSDs. Currently `rook osd` command implements this functionality.
 
 ### Create new OSDs
-| Sequence |Rook Operator  | OSD Provisioner   | Ceph OSD Deployment | 
-|---|---|---|---|
-| 1  | Read devices from cluster CRD and create a Pod for OSD Provisioner for each storage node  |   |   |
-| 2  | Watch Configmap  | Prepare OSDs  |   |
-| 3  | Watch Configmap  |  Persist OSD ID, datapath, and node info in a per node Configmap | |
-| 4  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | |
-| 5  | Create one deployment per OSD | |
-| 6  | | | Start `ceph-osd` Daemon one Pod per device |
+| Sequence |Rook Operator | OSD Discovery  | OSD Provisioner   | Ceph OSD Deployment | 
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create an OSD Discovery deployment on each storage node  |   |   |
+| 2  | Watch Configmap  | Discover OSD devices that fit those defined in Cluster CRD and update Configmap with the device paths| | |
+| 3  | Detect Configmap change, parse Configmap, extract device paths, and create an OSD Provisioner deployment for each device  |Detect device additions or removals | | |
+| 4  | Watch Configmap |Detect device additions or removals | Prepare OSDs, Persist OSD ID, datapath, and node info in a per node Configmap | |
+| 5  | Detect Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | Detect device additions or removals | | |
+| 6  | Create one deployment per OSD |Detect device additions or removals | |
+| 7  | |Detect device additions or removals | | Start `ceph-osd` Daemon one Pod per device |
 
 This change addresses the above limitations in the follow ways:
 - High reliability. Each `ceph-osd` daemon runs its own Pod, their restart and upgrade are by Kubernetes controllers. Upgrading OSD Provisioner Pod no longer restarts `ceph-osd` daemons.
-- More efficient resource requests. Once OSD Provisioner discovers and prepares all OSD devices, Rook Operator is informed of the topology and assigns appropriate resources to each Ceph OSD deployment.
+- More efficient resource requests. Once OSD Discovery detects all OSD devices, Rook Operator is informed of the topology and assigns appropriate resources to each Ceph OSD deployment.
+
+### Discussions
+
+It is expected OSD Discovery process will be merged into Rook Operator once local PVs are supported in Rook Cluster CRD. Rook Operator can infer the device topoloyg from local PV Configmaps. 
```

Comment:
for some period of time I would expect we need to support both raw devices and local pvs